crypto/evp/mac_lib.c

```diff
@@ -120,15 +120,14 @@ int EVP_MAC_update(EVP_MAC_CTX *ctx, const unsigned char *data, size_t datalen)
 int EVP_MAC_final(EVP_MAC_CTX *ctx,
                   unsigned char *out, size_t *outl, size_t outsize)
 {
-    int l = EVP_MAC_size(ctx);
+    size_t l = EVP_MAC_size(ctx);
+    int res = 1;
 
-    if (l < 0)
```

Comment:
It looks like EVP_MAC_size() cant be negative now - so this removal should be ok..

Reply:
It returns a _size_t_ so could never be negative.  It was being assigned into an _int_ and the negative check was for overflow.  This cannot happen now since the assignment is now to a _size_t_.